R/plot.R

```diff
@@ -42,92 +44,45 @@ plot.SLOPE = function(x,
   p <- NROW(coefs) # number of features
   m <- NCOL(coefs) # number of responses
 
-  args <- list()
-
   x <- switch(x_variable,
-              alpha = object$alpha,
-              deviance_ratio = object$deviance_ratio,
-              step = seq_along(object$alpha))
+              alpha = object[["alpha"]],
```

Comment:
I've got this habit because of foresight tbh.

Unlike `[[`, `$` uses `exact=FALSE` by default. If you run

`x <- list(a = 10, aaa = 100)`
`x$aa `
`x[["aa"]]`

you will get `[1] 100`, and `NULL`, respectively. In this case, the second outcome makes more sense, so using `$` you may generate some bugs (which are hard to be traced in more complicated code). The usage of `[[` is more general - if you want to use partial matching, you could equally run `x[["aa", exact = FALSE]]` and it seems to be a more deliberate way to do that.

Reply:
Ah, I see. Yeah I agree that `[[` is a better one-stop solution.